src/mainClass.js

```diff
@@ -2,44 +2,76 @@ import { getData, getAllCategories } from "./source";
 
 export class AllItems {
 
-  constructor(searchCategory, searchLinks, displayLinksForChosenCategory, searchChosenLinksOnly) {
+  constructor(searchCategory, searchLinks, displayLinksForChosenCategory) {
     this.categories = [];
     this.onReceivedCategories = searchCategory;
     this.displayChosenLinks = displayLinksForChosenCategory;
     this.links = [];
     this.onReceivedLinks = searchLinks;
-    this.chosenLinks = [];
-    this.onReceivedChosenLinks = searchChosenLinksOnly;
+    this.storageLinks = JSON.parse(localStorage.getItem('links'));
+    this.storageCategories = JSON.parse(localStorage.getItem('categories'));
+    this.filtered;
   }
 
-  setOnCategoriesReceived(onReceivedCategories) { 
-    this.onReceivedCategories = onReceivedCategories 
+  setOnCategoriesReceived(onReceivedCategories) {
+    this.onReceivedCategories = onReceivedCategories
   }
 
   setOnLinksReceived(onReceivedLinks) {
     this.onReceivedLinks = onReceivedLinks;
   }
 
-  setOnChosenLinksReceived(onReceivedChosenLinks) {
-    this.onReceivedChosenLinks = onReceivedChosenLinks;
-  }
-
   setDisplayChosenLinks(displayChosenLinks) {
     this.displayChosenLinks = displayChosenLinks;
   }
 
+  updateStorageLinks() {
+    localStorage.setItem('links', JSON.stringify(this.links));
+  }
+
+  updateStorageCategories() {
+    localStorage.setItem('categories', JSON.stringify(this.categories));
+  }
+
+  markAsImportant(el) {
+    if (el.important === false) {
+      el.important = true;
+    } else {
+      el.important = false;
+    }
+  }
+
+  setFiltered(param) {
+    this.filtered = param;
+  }
+
+  getAllLinks(filtered, e) {
+    if (filtered !== null) {
+      let filteredCategoriesAfterSearch = this.links.filter(el => {
+        if (el.API.toLowerCase().includes(e.target.value.toLowerCase())) {
+          return el;
+        }
+      })
+      return filteredCategoriesAfterSearch;
+    } else {
+      let filteredCategoriesAfterSearch = this.links.filter(el => {
+        if (el.API.toLowerCase().includes(e.target.value.toLowerCase())) {
```

Comment:
filtering again, I not understand?

Reply:
First outcome of If statement is filter in case that we have textFilter, because if we do, that means that this.links is filled with only objects that contains textFilter as their category. 

If textFilter is null, then this.links will be filled with all objects, so i am able to search trough links even if i dont have selected category.

Reply:
My logic is next: If we don't have category selected, we recive all links, if we have we send that param in API. Filtering by text is separate thing, so I think we don't need to connect that two things. So basic logic category selected or not links array is received proper and filtering by text is separate, if text filter is empty we return whole current links array no metter if selected category or not, because API server will filter for us it.